Library/OcCpuLib/OcCpuLib.c

```diff
@@ -416,10 +483,15 @@ ScanIntelProcessor (
       Recalculate
       );
 
+    //
+    // Determine the TSC frequency from Apple Platform Info.
+    //
+    Cpu->CPUFrequencyFromApple = InternalCalculateTSCFromApplePlatformInfo (NULL);
```

Comment:
Please do it only if `Cpu->CPUFrequencyFromART == 0 || Recalculate` is TRUE, in my opinion.